include/tvm/meta_schedule/database.h

```diff
@@ -325,6 +362,25 @@ class PyDatabaseNode : public DatabaseNode {
     return f_get_all_tuning_records();
   }
 
+  Optional<TuningRecord> QueryTuningRecord(const IRModule& mod, const Target& target,
+                                           const String& workload_name) final {
+    ICHECK(f_query_tuning_record != nullptr)
+        << "PyDatabase's QueryTuningRecord method not implemented!";
+    return f_query_tuning_record(mod, target, workload_name);
```

Comment:
Note that `QueryXYZ`  does have their default implementation: https://github.com/apache/tvm/blob/2af9b90ec191424724842795c552d4c15682eb8c/src/meta_schedule/database/database.cc#L159. Therefore, if `f_query_tuning_record` is null, we should still use the default implementation rather than error out.

Let's also have both cases tests (has default impl vs no default impl)